src/BodyParams/BodyParamsMiddleware.php

```diff
@@ -82,10 +83,17 @@ public function __invoke(ServerRequestInterface $request, ResponseInterface $res
             }
 
             // Matched! Parse and pass on to the next
-            return $next(
-                $strategy->parse($request),
-                $response
-            );
+            try {
+                return $next(
+                    $strategy->parse($request),
+                    $response
+                );
+            } catch (MalformedRequestBodyException $exception) {
```

Comment:
I'm thinking it may make more sense to return a response, instead of invoking `$next`; essentially, the attempt to parse the body has failed at this point, which means we have a malformed request.